llvm_passes/HipLowerMemset.cpp

```diff
@@ -0,0 +1,71 @@
+//===- HipLowerMemset.cpp -------------------------------------------------===//
+//
+// Part of the chipStar Project, under the Apache License v2.0 with LLVM
+// Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Expand llvm.memset intrinsics to loops before handing the bitcode to
+// llvm-spirv translator. The translator expands memset intrinsics to emulation
+// function with a name encoding the destination pointer type
+// (e.g. "spirv.spirv.llvm_memset_p0_i32..."). During reverse translation,
+// llvm-spirv attempts to recreate the original memset intrinsic which can be
+// incorrect if we are jumping from opaque pointer world to typed pointer world
+// or vice versa. LLVM module verification may fail with the following message:
```

Comment:
Is this a SPIR-V translator bug? Can you add a link to the issue here?

Reply:
I'll file the issue later.